lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1598 @@
+                eg_eni.associate_public_ip_address = eni.get('associate_public_ip_address')
+
+            if eni.get('delete_on_termination') is not None:
+                eg_eni.delete_on_termination = eni.get('delete_on_termination')
+
+            if eni.get('groups') is not None:
+                eg_eni.groups = eni['groups']
+
+            if eni.get('network_interface_id') is not None:
+                eg_eni.network_interface_id = eni.get('network_interface_id')
+
+            if eni.get('private_ip_address') is not None:
+                eg_eni.private_ip_address = eni.get('private_ip_address')
+
+            if eni.get('subnet_id') is not None:
+                eg_eni.subnet_id = eni.get('subnet_id')
+
+            if eni.get('associate_ipv6_address') is not None:
+                eg_eni.associate_ipv6_address = eni.get('associate_ipv6_address')
+
+            expand_private_ip_addresses(eg_eni, eni)
+
+            eg_enis.append(eg_eni)
+
+        if eg_enis.__sizeof__() > 0:
+            eg_launchspec.network_interfaces = eg_enis
+
+
+def expand_private_ip_addresses(eg_eni, eni):
+    if eni.get('private_ip_addresses') is not None:
+        eg_pias = []
+        pias = eni.get('private_ip_addresses')
+
+        for pia in pias:
+            eg_pia = spotinst.aws_elastigroup.PrivateIpAddress()
+
+            eg_pia_address = pia.get('private_ip_address')
+            eg_pia_primary = pia.get('primary')
+            eg_pia.private_ip_address = eg_pia_address
+            eg_pia.primary = eg_pia_primary
+
+            eg_pias.append(eg_pia)
+
+        eg_eni.private_ip_addresses = eg_pias
+
+
+def expand_persistence(eg_strategy, persistence):
+    if persistence is not None:
+        eg_persistence = spotinst.aws_elastigroup.Persistence()
+        eg_persistence.should_persist_root_device = persistence.get('should_persist_root_device')
+        eg_persistence.should_persist_block_devices = persistence.get('should_persist_block_devices')
+        eg_persistence.should_persist_private_ip = persistence.get('should_persist_private_ip')
+        eg_strategy.persistence = eg_persistence
+
+
+def expand_scaling(eg, module):
+    up_scaling_policies = module.params['up_scaling_policies']
+    down_scaling_policies = module.params['down_scaling_policies']
+
+    eg_scaling = spotinst.aws_elastigroup.Scaling()
+
+    if up_scaling_policies is not None:
+        eg_up_scaling_policies = expand_scaling_policies(up_scaling_policies)
+        if eg_up_scaling_policies.__sizeof__() > 0:
+            eg_scaling.up = eg_up_scaling_policies
+
+    if down_scaling_policies is not None:
+        eg_down_scaling_policies = expand_scaling_policies(down_scaling_policies)
+        if eg_down_scaling_policies.__sizeof__() > 0:
+            eg_scaling.down = eg_down_scaling_policies
+
+    if eg_scaling.down is not None or eg_scaling.up is not None:
+        eg.scaling = eg_scaling
+
+
+def expand_scaling_policies(scaling_policies):
```

Comment:
Generally I'm not so happy with this long list of if statements.  What I would rather have is a set of named objects (like the `eg.policy`) and a set of lists of relevant parameters (`eg_policy_params=['policy_name', 'namespace....`) and then iterate through the list to build the object from the original policy object.  I think this would be easier to maintain in future.

Reply:
Great idea. I have implemented this as you suggested. In some cases the spotinst and module name are not identical to conform to aws usage which users are familiar with, as is the case with min_size (As opposed to 'minimum' in spotinst). For these cases I have also allowed to provide a dictionary in the list with the values for the spotinst and ansible names, and have taken it into account in the expand_fields method which populates the objects.